src/Model/Post/Engagement.php

```diff
@@ -77,6 +65,15 @@ public static function storeFromItem(array $item)
 			$store = Contact::hasFollowers($parent['owner-id']);
 		}
 
+		if (!$store) {
+			$tagList = Relay::getSubscribedTags();
+			foreach (array_column(Tag::getByURIId($item['parent-uri-id'], [Tag::HASHTAG]), 'name') as $tag) {
+				if (in_array($tag, $tagList)) {
+					$store = true;
```

Comment:
You can exit early from the `foreach` if the condition is met.
```suggestion
					$store = true;
					break;
```